src/lib/smartthings/smartappService.js

```diff
@@ -0,0 +1,135 @@
+'use strict';
+
+const request = require('request-promise-native'),
+      smartapp = require('../smartApp');
+
+const smartappService = {
+    getSmartAppContext: async function(installedSmartAppId) {
+        var contextError;
+        const smartAppContext = await smartapp.withContext(installedSmartAppId)
+        .catch((err) => {
```

Comment:
Can a `try` `catch` be used here?

Reply:
It can yeah, but this does the same thing. I tend to like this format better because it avoids extra nesting, and multiple `await`s can be done in a row without a nasty looking long series of try/catches.

In other projects I use a wrapper function that makes it really nice looking but I think that obfuscation might confuse people who don't pay attention to the wrapper being called.